### PR TITLE
Port assert to expect in tests.

### PR DESCRIPTION
**Pull request by WiXSL**

### CODE REVIEW

File: packages/ra-core/src/controller/field/ReferenceManyFieldController.spec.tsx [hunk 1]
```diff
@@ -1,9 +1,20 @@
...
+// Ignore warnings about act()
+// See https://github.com/testing-library/react-testing-library/issues/281,
+// https://github.com/facebook/react/issues/14769
+const originalError = console.error;
+jest.spyOn(console, 'error').mockImplementation((...args) => {
+    if (/Warning.*not wrapped in act/.test(args[0])) {
+        return;
+    }
+    originalError.call(console, ...args);
+});
```
Comment (djhi) [inline packages/ra-core/src/controller/field/ReferenceManyFieldController.spec.tsx:16]: I'm not sure we should do this. We are probably not waiting for side effects to apply where we should.

Comment (WiXSL) [reply packages/ra-core/src/controller/field/ReferenceManyFieldController.spec.tsx]: Done!

Comment (djhi) [reply packages/ra-core/src/controller/field/ReferenceManyFieldController.spec.tsx]: If you have time and noticed those warnings, would you mind trying to add the wait/expect calls needed to make them go away ?

Comment (WiXSL) [reply packages/ra-core/src/controller/field/ReferenceManyFieldController.spec.tsx]: Ok. I don't get any warnings now!

File: packages/ra-core/src/dataProvider/fetch.spec.ts [hunk 2]
```diff
@@ -79,11 +78,11 @@ describe('createHeadersFromOptions', () => {
         };
 
         const headersWithMethod = createHeadersFromOptions(optionsWithMethod);
-        assert.strictEqual(headersWithMethod.get('Content-Type'), null);
+        expect(headersWithMethod.get('Content-Type')).toStrictEqual(null);
```
Comment (fzaninotto) [inline packages/ra-core/src/dataProvider/fetch.spec.ts:81]: toBeNull? Same for the next one

File: packages/ra-core/src/reducer/admin/notifications.spec.ts [hunk 3]
```diff
@@ -8,11 +8,12 @@ import reducer from './notifications';
 
 describe('notifications reducer', () => {
     it('should return empty notification by default', () => {
-        assert.deepEqual([], reducer(undefined, { type: 'foo' }));
+        expect([]).toEqual(reducer(undefined, { type: 'foo' }));
```
Comment (fzaninotto) [inline packages/ra-core/src/reducer/admin/notifications.spec.ts:11]: I suggest we swap the two operands here, I'm not a fan of this type of assertion where the result is put first.

File: packages/ra-core/src/reducer/admin/ui.spec.ts [hunk 4]
```diff
@@ -13,52 +13,47 @@ describe('ui reducer', () => {
         viewVersion: 0,
     };
     it('should return hidden sidebar by default', () => {
-        assert.deepEqual(defaultState, reducer(undefined, { type: 'foo' }));
+        expect(defaultState).toEqual(reducer(undefined, { type: 'foo' }));
```
Comment (fzaninotto) [inline packages/ra-core/src/reducer/admin/ui.spec.ts:16]: same here, we should inverse operands

File: packages/ra-core/src/reducer/admin/notifications.spec.ts (hunk 3 above)
Comment (WiXSL) [reply packages/ra-core/src/reducer/admin/notifications.spec.ts]: Done!

File: packages/ra-core/src/reducer/admin/ui.spec.ts (hunk 4 above)
Comment (WiXSL) [reply packages/ra-core/src/reducer/admin/ui.spec.ts]: Done!

File: packages/ra-core/src/dataProvider/fetch.spec.ts (hunk 2 above)
Comment (WiXSL) [reply packages/ra-core/src/dataProvider/fetch.spec.ts]: Done!

**Comment by fzaninotto** [comment]
Thanks!

**Pull request merged**